jaxrs/src/test/scala/com/cerner/beadledom/jaxrs/exceptionmappers/BaseExceptionMapperSpec.scala

```diff
@@ -0,0 +1,31 @@
+package com.cerner.beadledom.jaxrs.exceptionmappers
+
+import com.cerner.beadledom.jaxrs.models.JsonError
+
+import com.google.gson.Gson
+
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FunSpec, ShouldMatchers}
+
+import play.api.libs.json.{JsValue, Json}
+
+import javax.ws.rs.core.Response.Status
+
+/**
+  * Base spec to be used by ExceptionMapper spec classes.
+  *
+  * @author Cal Fisher
+  */
+class BaseExceptionMapperSpec
+    extends FunSpec with ShouldMatchers with BeforeAndAfterEach with MockitoSugar {
+
+  val gson = new Gson
+
+  def createJsonError(status: Status): JsValue = {
```

Comment:
Rather than using this method and using Gson you could have custom error matchers to use in the tests. @b-boogaard already wrote a matcher that we have been using. It's probably not complete for all of the cases needing to be tested with these exception mappers, but it should be easy to expand.

The matcher could go in the `beadledom-testing` module.

```scala
/**
 * [JsonError] matchers.
 *
 * @author Brian van de Boogaard
 */
trait JsonErrorMatchers {
  val mapper = new ObjectMapper()

  class JsonErrorResponse(expectedStatus: Int, expectedMessage: String) extends Matcher[String] {
    def apply(left: String) = {
      val jsonError = mapper.readValue(left, classOf[JsonError])
      val code = jsonError.code()
      val message = jsonError.message()
      MatchResult(
        code == expectedStatus && message.equals(expectedMessage),
        s"""expected: [code: $expectedStatus, message: $expectedMessage] actual [code: $code, message: $message]""",
        s"""expected: [code: $expectedStatus, message: $expectedMessage] matched [code: $code, message: $message]"""
      )
    }
  }

  class BadRequestMatcher(expectedMessage: String) extends JsonErrorResponse(400, expectedMessage)
  class NotFoundMatcher(expectedMessage: String) extends JsonErrorResponse(404, expectedMessage)

  def beBadRequestError(expectedMessage: String = Response.Status.BAD_REQUEST
      .getReasonPhrase) = new BadRequestMatcher(expectedMessage)
  def beNotFoundError(
      expectedMessage: String = Response.Status.NOT_FOUND.getReasonPhrase) = new NotFoundMatcher(
    expectedMessage)
  def beJsonError(expectedStatus: Int, expectedMessage: String) = new JsonErrorResponse(
    expectedStatus, expectedMessage)
}

object JsonErrorMatchers extends JsonErrorMatchers
```

Reply:
I attempted to add these custom error matchers to the `beadledom-testing` module, but ran into a cyclical dependency issue because the `JsonError` class lives in the `beadledom-jaxrs` module, and the `beadledom-jaxrs` module would be dependent on `beadledom-testing` for these custom error matchers.

Should the `JsonError` and `ErrorDetail` models be moved to a different place?

Reply:
Good catch.

I had been thinking the `JsonError` and `ErrorDetail` classes should go somewhere else I just hadn't yet formulated where.

I had an offline discussion with Brian and we are thinking they could go in a new module named `beadledom-json-common` with a directory named just `json-common`.

Reply:
I moved the `JsonError` and `ErrorDetail` classes to a new module named `beadledom-json-common` with a directory name `json-common`: https://github.com/cerner/beadledom/pull/45/commits/b3f0569735e58069bdbc0a3a759fca76a905db1f

I have also added the `JsonErrorMatchers` class above and expanded on the types of exceptions beind tested: https://github.com/cerner/beadledom/pull/45/commits/5526d8c027550e6ea5657d742f4ae878f66ac4d7#diff-09a50fae7f92d0c897a4330007129927R1

Consumed the new `JsonErrorMatchers` in unit tests: https://github.com/cerner/beadledom/pull/45/commits/5526d8c027550e6ea5657d742f4ae878f66ac4d7#diff-dd19a59445cd85b525abe410a27416d4R61